admin/data/sql/migrations/20190409201024.do.alter-check-window-date-types.sql

```diff
@@ -0,0 +1,30 @@
+DECLARE @columnName VARCHAR(max)
+DECLARE @sql NVARCHAR(max)
+
+DECLARE cur CURSOR FOR
+ SELECT COLUMN_NAME
+   FROM INFORMATION_SCHEMA.COLUMNS
+  WHERE COLUMN_NAME LIKE '%Date'
+    AND TABLE_NAME = 'checkWindow'
+
+OPEN cur
+
+FETCH NEXT FROM cur INTO @columnName;
+
+WHILE @@fetch_status = 0
+BEGIN
+    SET @sql = '
+    DECLARE @ObjectName NVARCHAR(100)
+    SELECT @ObjectName = OBJECT_NAME([default_object_id]) FROM SYS.COLUMNS
+    WHERE [object_id] = OBJECT_ID(''[mtc_admin].[checkWindow]'') AND [name] = ' + QUOTENAME(@columnName,'''') + '
+    EXEC(''ALTER TABLE [mtc_admin].[checkWindow] DROP CONSTRAINT '' + @ObjectName)
+    ALTER TABLE [mtc_admin].checkWindow ALTER COLUMN ' + @columnName + ' DATETIME2 NOT NULL
+    ALTER TABLE [mtc_admin].checkWindow ADD DEFAULT GETUTCDATE() FOR ' + @columnName + '
+    '
```

Comment:
This is over engineered and makes it very difficult to understand your intentions.

Please just add explicit statements for each column change

Reply:
@GuyHarwood it was quite convenient to have this instead of 6 groups of queries. The intention was to get the start and end date column names from the check window table and for each one:
- drop the unnamed constraint
- modify the type to datetime2
- add constraint for default value

I can agree that it seems a bit over-engineered so I'll make these explicit.

Reply:
@GuyHarwood updated

Reply:
There is a misunderstanding here - there is absolutely no point in querying out the column metadata.  Just perform the column changes explicitly, one by one with statically typed names